plugins/vkdemo/plugin.cpp

```diff
@@ -0,0 +1,983 @@
+        };
+
+        VmaAllocationCreateInfo staging_alloc_info{};
+        staging_alloc_info.flags = VMA_ALLOCATION_CREATE_HOST_ACCESS_SEQUENTIAL_WRITE_BIT;
+        staging_alloc_info.usage = VMA_MEMORY_USAGE_AUTO;
+
+        VkBuffer      staging_buffer;
+        VmaAllocation staging_buffer_allocation;
+        VK_ASSERT_SUCCESS(vmaCreateBuffer(vma_allocator, &staging_buffer_info, &staging_alloc_info, &staging_buffer,
+                                          &staging_buffer_allocation, nullptr))
+
+        VkBufferCreateInfo buffer_info{
+            VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO,                                // sType
+            nullptr,                                                             // pNext
+            0,                                                                   // flags
+            sizeof(indices[0]) * indices.size(),                                 // size
+            VK_BUFFER_USAGE_TRANSFER_DST_BIT | VK_BUFFER_USAGE_INDEX_BUFFER_BIT, // usage
+            {},                                                                  // sharingMode
+            0,                                                                   // queueFamilyIndexCount
+            nullptr                                                              // pQueueFamilyIndices
+        };
+
+        VmaAllocationCreateInfo alloc_info{};
+        alloc_info.usage = VMA_MEMORY_USAGE_GPU_ONLY;
+
+        VmaAllocation buffer_allocation;
+        VK_ASSERT_SUCCESS(vmaCreateBuffer(vma_allocator, &buffer_info, &alloc_info, &index_buffer, &buffer_allocation, nullptr))
+
+        void* mapped_data;
+        VK_ASSERT_SUCCESS(vmaMapMemory(vma_allocator, staging_buffer_allocation, &mapped_data))
+        memcpy(mapped_data, indices.data(), sizeof(indices[0]) * indices.size());
+        vmaUnmapMemory(vma_allocator, staging_buffer_allocation);
+
+        VkCommandBuffer command_buffer_local = vulkan_utils::begin_one_time_command(ds->vk_device, command_pool);
+        VkBufferCopy    copy_region{
+            0,                                  // srcOffset
+            0,                                  // dstOffset
+            sizeof(indices[0]) * indices.size() // size
+        };
+        vkCmdCopyBuffer(command_buffer_local, staging_buffer, index_buffer, 1, &copy_region);
+        vulkan_utils::end_one_time_command(ds->vk_device, command_pool, ds->graphics_queue, command_buffer_local);
+
+        vmaDestroyBuffer(vma_allocator, staging_buffer, staging_buffer_allocation);
+    }
+
+    void create_pipeline(VkRenderPass render_pass, uint32_t subpass) {
+        if (pipeline != VK_NULL_HANDLE) {
+            throw std::runtime_error("timewarp_vk::create_pipeline: pipeline already created");
+        }
+
+        auto           folder = std::string(SHADER_FOLDER);
+        VkShaderModule vert =
+            vulkan_utils::create_shader_module(ds->vk_device, vulkan_utils::read_file(folder + "/demo.vert.spv"));
```

Comment:
`SHADER_FOLDER` no longer points to a valid directory. The system crashes on launch.

Reply:
I have fixed this. The shader files are now installed in $CMAKE_INSTALL_PREFIX/share/illixr/shaders and the proper location is given to the vkdemo and timewarp_vk code at compilation time. Just be sure to run make install